frontend/src/scenes/settings/project/SessionRecordingSettings.tsx

```diff
@@ -206,165 +206,164 @@ export function ReplayCostControl(): JSX.Element {
+                            updateCurrentTeam({ session_recording_minimum_duration_milliseconds: v })
+                        }}
+                        options={SESSION_REPLAY_MINIMUM_DURATION_OPTIONS}
+                        value={currentTeam?.session_recording_minimum_duration_milliseconds}
+                    />
+                </div>
+                <p>
+                    Setting a minimum session duration will ensure that only sessions that last longer than that value
+                    are collected. This helps you avoid collecting sessions that are too short to be useful.
+                </p>
+            </>
+            <>
+                <div className={'flex flex-col space-y-2'}>
+                    <LemonLabel className="text-base">Enable recordings using feature flag</LemonLabel>
+                    <div className={'flex flex-row justify-start'}>
+                        <FlagSelector
+                            value={currentTeam?.session_recording_linked_flag?.id ?? undefined}
+                            onChange={(id, key) => {
+                                updateCurrentTeam({ session_recording_linked_flag: { id, key } })
                             }}
-                            options={SESSION_REPLAY_MINIMUM_DURATION_OPTIONS}
-                            value={currentTeam?.session_recording_minimum_duration_milliseconds}
+                            readOnly={!costControlFeaturesEnabled}
+                            readOnlyReason="Please contact support to enable this feature."
                         />
-                    </div>
-                    <p>
-                        Setting a minimum session duration will ensure that only sessions that last longer than that
-                        value are collected. This helps you avoid collecting sessions that are too short to be useful.
-                    </p>
-                </>
-            )}
-            {(featureFlags[FEATURE_FLAGS.SESSION_RECORDING_SAMPLING] || featureFlagRecordingFeatureEnabled) && (
-                <>
-                    <div className={'flex flex-col space-y-2'}>
-                        <LemonLabel className="text-base">Enable recordings using feature flag</LemonLabel>
-                        <div className={'flex flex-row justify-start space-x-2'}>
-                            <FlagSelector
-                                value={currentTeam?.session_recording_linked_flag?.id ?? undefined}
-                                onChange={(id, key) => {
-                                    updateCurrentTeam({ session_recording_linked_flag: { id, key } })
-                                }}
+                        {currentTeam?.session_recording_linked_flag && (
+                            <LemonButton
+                                disabledReason={
+                                    !costControlFeaturesEnabled && 'Please contact support to enable this feature.'
+                                }
```

Comment:
Might be nit-picking... and I think maybe an unlikely state for a real user to be in. But I think if you've got a linked flag and you're not allowed to use the feature we should still let you clear the flag...